spec/consensus/proposer-based-timestamp/pbts-sysmodel_002_draft.md

```diff
@@ -79,172 +86,264 @@ defining a lower bound, a *minimum time* that a correct process assigns to propo
-- `p` proposes a valid value `v` and sets `v.time` to the time it reads from its local clock,
+then there is a correct process `p` (not necessarily the same above considered) such that:
 
-Then let `q` be a correct process that receives `p`'s proposal, we have:
+- `beginRound(p,v.round) <= proposalReceptionTime(p,v.round) <= beginRound(p,v.round+1)` and
+- `proposalReceptionTime (p,v.round) - MSGDELAY - PRECISION <= v.time <= proposalReceptionTime(p,v.round) + PRECISION`
 
-- `q` receives `p`'s proposal after its clock reads `v.time - PRECISION`, and
-- if `q` is at or joins round `r` while `p`'s proposal is being transmitted,
-then `q` receives `p`'s proposal before its clock reads `v.time + MSGDELAY + PRECISION`
+That is, a correct process `p` started round `v.round` and, while still at
+round `v.round`, received a `PROPOSAL` message from round `v.round` proposing
+`v`.
+Moreover, the reception time of the original proposal for `v`, according with
+`p`'s local clock, enabled `p` to consider the proposal time `v.time` as
+`timely`.
+This is the requirement established by PBTS for issuing a `PREVOTE` for the
+proposal `(v, v.time, v.round)`, so for the eventual decision of `v`.
 
-> Note that, before `GST`, we cannot ensure that every correct process receives `p`'s proposals, nor that it does it while ready to accept a round `r` proposal.
+### Liveness
 
-A correct process `q` as above defined must then consider `p`'s proposal `timely`.
-It will then broadcast a `PREVOTE` message for `v` at round `r`,
-thus enabling, from the Time-Validity point of view, `v` to be eventually decided.
+The liveness of PBTS relies on correct processes accepting proposal times
+assigned by correct proposers.
+We thus present a set of conditions for assigning a proposal time `v.time` so
+that every correct process should be able to issue a `PREVOTE` for `v`.
 
-#### Under-estimated `MSGDELAY`s
+#### **[PBTS-LIVENESS.0]**
 
-The liveness assumptions of PBTS are conditioned by a conservative and clever
-choice of the timing parameters, specially of `MSGDELAY`.
-In fact, if the transmission delay for a message carrying a proposal is wrongly
-estimated, correct processes may never consider a valid proposal as `timely`.
+If
 
-To circumvent this liveness issue, which could result from a misconfiguration,
-we assume that the `MSGDELAY` parameter can be increased as rounds do not
-succeed on deciding a value, possibly because no proposal is considered
-`timely` by enough processes.
-The precise behavior for this workaround is under [discussion](https://github.com/tendermint/spec/issues/371).
+- the proposer of a round `r` of consensus is correct
+- and it proposes a value `v` for the first time, with associated proposal time `v.time`
+
+then the proposal `(v, v.time, r)` is accepted by every correct process provided that:
+
+- `min{p is correct : beginRound(p,r)} <= v.time <= max{p is correct : beginRound(p,r)}` and
```

Comment:
Is `beginRound` here the 'global' time or the local time?
From the above definition:
```
to represent the time the process `p` reads from its clock
when it starts round `r` of consensus.
```

I'm a bit unclear if it is the global time _at which_ it reads its clock or the value that it reads from its clock.


It I'm understanding correctly, this is basically saying that `v.time` must be between the times at which the processes enter the round.

Reply:
`beginRound` is local time. The time a process reads from its clock is the time "displayed" by its clock.

I will consider a less ambiguous way to reference to the local time.

Reply:
Yeah, that sentence is a little bit ambiguous. Maybe, `the value of process `p`'s local clock when it starts round `r` of consensus`. The other way is likely ok, but maybe a little tough to parse.